File: examples/personal_assistant.py
from mcpscraper import (
    AssistantApprovalDecision,
    AssistantCursor,
    AssistantIdempotencyKey,
    McpToolsClient,
)


def submit_command(
    client: McpToolsClient,
    assistant_ref: str,
    idempotency_key: AssistantIdempotencyKey,
) -> None:
    client.assistant.command(
        assistant_ref=assistant_ref,
        instruction="Draft a reply, but do not send it without my approval.",
        idempotency_key=idempotency_key,
    )


def list_pending_approvals(
    client: McpToolsClient,
    cursor: AssistantCursor | None = None,
) -> None:
    client.assistant.approvals_list(state="pending", cursor=cursor, page_size=25)


def decide_approval(
    client: McpToolsClient,
    decision: AssistantApprovalDecision,
) -> None:
    client.assistant.approval_decide(**decision.model_dump(by_alias=True))

File: examples/personal-assistant.ts
import {
  McpToolsClient,
  type AssistantApprovalDecision,
  type AssistantCursor,
  type AssistantIdempotencyKey,
} from '../packages/memory/src/index.js'

export async function submitAndReview(
  client: McpToolsClient,
  assistantRef: string,
  idempotencyKey: AssistantIdempotencyKey,
): Promise<void> {
  await client.assistant.command({
    assistantRef,
    instruction: 'Draft a reply, but do not send it without my approval.',
    idempotencyKey,
  })

  let cursor: AssistantCursor | undefined
  do {
    const page = await client.assistant.approvalsList({ state: 'pending', cursor, pageSize: 25 })
    cursor = typeof page.data === 'object' && page.data !== null && 'nextCursor' in page.data
      ? String(page.data.nextCursor)
      : undefined
  } while (cursor)
}

export async function decideApproval(
  client: McpToolsClient,
  decision: AssistantApprovalDecision,
): Promise<void> {
  await client.assistant.approvalDecide(decision)
}
